common/src/main/java/org/apache/drill/exec/metrics/CpuGaugeSet.java

```diff
@@ -32,13 +32,24 @@
  */
 @SuppressWarnings("restriction")
 public class CpuGaugeSet implements MetricSet {
+  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(CpuGaugeSet.class);
 
   private OperatingSystemMXBean osMXBean;
   private RuntimeMXBean rtMXBean;
+  private static String nonHotSpotJVMName;
 
   public CpuGaugeSet() {
-    this.osMXBean = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
+    //DRILL-6702: Instead of worrying about compiling with IBM JDK, for now, we shall provide no CPU metrics for non-HotSpot JVMs
+    if (System.getProperty("java.vm.name").contains("HotSpot")) {
+      this.osMXBean = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
+    } else {
+      this.osMXBean = null;
+      nonHotSpotJVMName = System.getProperty("java.vm.name");
+    }
     this.rtMXBean = ManagementFactory.getRuntimeMXBean();
+    if (nonHotSpotJVMName != null) {
+      logger.warn("Detected non-HotSpot JVM [{}]. CPU Metrics in the WebUI will not be available!", nonHotSpotJVMName);
+    }
```

Comment:
```
public class CpuGaugeSet implements MetricSet {
  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(CpuGaugeSet.class);

  private final RuntimeMXBean rtMXBean;
  private final OperatingSystemMXBean osMXBean;

  public CpuGaugeSet() {
    //DRILL-6702: Instead of worrying about compiling with IBM JDK, for now, we shall provide no CPU metrics for non-HotSpot JVMs
    this.rtMXBean = ManagementFactory.getRuntimeMXBean();
    java.lang.management.OperatingSystemMXBean operatingSystemMXBean = ManagementFactory.getOperatingSystemMXBean();
    if (operatingSystemMXBean instanceof OperatingSystemMXBean) {
      this.osMXBean = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
    } else {
      this.osMXBean = null;
      logger.warn("Detected non-HotSpot JVM [{}]. CPU Metrics in the WebUI will not be available!", System.getProperty("java.vm.name"));
    }
  }
...
```